janitor_test.go

```diff
@@ -0,0 +1,89 @@
+package vspherejanitor_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	vspherejanitor "github.com/travis-ci/vsphere-janitor"
+	"github.com/travis-ci/vsphere-janitor/mock"
+)
+
+type testCase struct {
+	now        time.Time
+	config     *vspherejanitor.JanitorOpts
+	vms        []*mock.VMData
+	poweredOff []string
+	destroyed  []string
+}
+
+var aTime = time.Date(2016, 01, 15, 12, 0, 0, 0, time.UTC)
+
+var janitorTestCases []testCase = []testCase{
+	{
+		now: aTime,
+		config: &vspherejanitor.JanitorOpts{
+			Cutoff:         time.Hour,
+			SkipDestroy:    false,
+			Concurrency:    1,
+			RatePerSecond:  100,
+			SkipZeroUptime: true,
+			SkipNoBootTime: true,
+		},
+		vms: []*mock.VMData{
+			{
+				Name:      "old-powered-on",
+				Uptime:    2 * time.Hour,
+				BootTime:  timePointer(aTime.Add(-2 * time.Hour)),
+				PoweredOn: true,
+			},
+			{
+				Name:      "new-powered-on",
+				Uptime:    30 * time.Minute,
+				BootTime:  timePointer(aTime.Add(-30 * time.Minute)),
+				PoweredOn: true,
+			},
+		},
+	},
+}
+
+func TestJanitor(t *testing.T) {
+	for _, c := range janitorTestCases {
+		vmLister := mock.NewVMLister(map[string][]*mock.VMData{
+			"/": c.vms,
+		})
+
+		janitor := vspherejanitor.NewJanitor(vmLister, c.config)
+		err := janitor.Cleanup(context.TODO(), "/")
+		assertOk(t, "janitor.Cleanup(/)", err)
+
+		assertEqual(t, `PoweredOff("/", "old-powered-on")`, true, vmLister.PoweredOff("/", "old-powered-on"))
+		assertEqual(t, `Destroyed("/", "old-powered-on")`, true, vmLister.Destroyed("/", "old-powered-on"))
+		assertEqual(t, `PoweredOff("/", "new-powered-on")`, true, vmLister.PoweredOff("/", "old-powered-on"))
+		assertEqual(t, `Destroyed("/", "new-powered-on")`, true, vmLister.Destroyed("/", "old-powered-on"))
```

Comment:
Mostly because they are tiny functions, and I didn't see the need to pull in all of testify for it. I can switch it out if you think it's better to use that, though.